src/WorkingCapital.sol

```diff
@@ -10,68 +10,123 @@ import {IHats} from "./../hatsprotocol/src/Interfaces/IHats.sol";
 
 contract WorkingCapital is PluginCloneable {
 
+
     bytes32 public constant UPDATE_SPENDING_LIMIT_PERMISSION_ID = keccak256('UPDATE_SPENDING_LIMIT_PERMISSION');
 
+    struct MyAction {
+        address to;
+        uint256 value;
+        address ERC20;
+    }
+
+
     IHats public hatsProtocolInstance;
-    uint256 public hatId; 
-    uint256 public spendingLimitETH;
+    uint256 public hatId;
+    mapping(address => uint256) public spendingLimit;
 
     uint private currentMonth;
     uint private currentYear;
-    uint256 private remainingBudget;
+    mapping(address => uint256) private remainingBudget;
+    address[] availableTokens;
 
     /// @notice Initializes the contract.
     /// @param _dao The associated DAO.
     /// @param _hatId The id of the hat.
-    function initialize(IDAO _dao, uint256 _hatId, uint256 _spendingLimitETH) external initializer {
+    function initialize(IDAO _dao, uint256 _hatId,address[] memory _token ,uint256[] memory _spendingLimit) external initializer {
         __PluginCloneable_init(_dao);
         hatId = _hatId;
         // TODO get this from environment per network (this is goerli)
         hatsProtocolInstance = IHats(0x3bc1A0Ad72417f2d411118085256fC53CBdDd137);
-        spendingLimitETH = _spendingLimitETH;
+        require(_token.length==_spendingLimit.length,"Length of token address and spendingLimit array is not equal");
+//        spendingLimit = _spendingLimit;
+        for (uint j=0; j < _token.length; j+=1) {
+            spendingLimit[_token[j]]=_spendingLimit[j];
+        }
+        availableTokens=_token;
+    }
+
+
+    /// @notice Check that the given token has allowance
+    /// @param _token check that given token has allowance
+    function isTokenAvailable(address _token) internal view returns (bool) {
+        for (uint256 i = 0; i < availableTokens.length; i++) {
+            if (availableTokens[i] == _token) {
+                return true;
+            }
+        }
+        return false;
     }
```

Comment:
I think instead of using for you can use hashtable. 
Instead of using availableTokens array just use spendingLimit[token].
if spendingLimit[token] was 0 or undefined it means that the token is not available.